template/user-data.tpl

```diff
@@ -12,11 +12,27 @@ EOL
 
 ${eip}
 
+echo ""
+echo "Installing AWS CLI..."
+AWSCLI_LOG=/var/log/aws-cli-install.log
+touch $AWSCLI_LOG
+
+for i in {1..7}; do
+  echo "Attempt: ---- " $i
+  curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64-${aws_cli_version}.zip" -o "awscliv2.zip" >>$AWSCLI_LOG 2>&1 && break || sleep 60
```

Comment:
This line means we can't use ARM instances to run gitlab-runner instance. These are usually cheaper and worth using.

`arch` command is available at least on Amazon AMIs, it may be worth probing. Perhaps something like

```diff
diff --git a/template/user-data.tpl b/template/user-data.tpl
index 6eac5b3..2775564 100644
--- a/template/user-data.tpl
+++ b/template/user-data.tpl
@@ -17,9 +17,15 @@ echo "Installing AWS CLI..."
 AWSCLI_LOG=/var/log/aws-cli-install.log
 touch $AWSCLI_LOG
 
+ARCH=$(arch)
+if [ $? -ne 0 ]; then
+  echo "Failed to detected machine architecture for AWS CLI."
+  exit 1
+fi
+
 for i in {1..7}; do
   echo "Attempt: ---- " $i
-  curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64-${aws_cli_version}.zip" -o "awscliv2.zip" >>$AWSCLI_LOG 2>&1 && break || sleep 60
+  curl "https://awscli.amazonaws.com/awscli-exe-linux-$${ARCH}-${aws_cli_version}.zip" -o "awscliv2.zip" >>$AWSCLI_LOG 2>&1 && break || sleep 60
 done
 unzip awscliv2.zip >>$AWSCLI_LOG 2>&1
 sudo ./aws/install >>$AWSCLI_LOG 2>&1
```